docs/adr-0001-publishing_metric_docs.md

```diff
@@ -0,0 +1,84 @@
+# Publishing documentation for metrics
+
+* Status: proposed <!-- [proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)] optional -->
+* Deciders: emtwo, scholtzan, tdsmith, TBD
+* Date: 2020-11-19
+
+Technical Story: https://github.com/mozilla/jetstream/issues/296
+
+## Context and Problem Statement
+
+We need to present some context for our metrics on dashboards and use friendlier names than the column labels we have in our output today.
+We've already decided that this context should live next to the metric definitions, either in mozanalysis or in jetstream-config.
+We haven't decided how to make them accessible to downstream consumers like Experimenter and Partybal yet.
+
+Jetstream is the point in the system that understands which metrics will be included in an analysis
+and has access to the context defined with the metrics,
+so it makes sense for Jetstream to be responsible for collating those definitions.
+
+## Decision Drivers
+
+* We should minimize the number of systems that downstream consumers have to contend with.
+* ?
+
+## Considered Options
+
+* dictionary.json
+* Google Data Catalog
+* Something else
+
+## Decision Outcome
+
+TK
+
+<!-- Chosen option: "[option 1]", because [justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+-->
+
+## Pros and Cons of the Options <!-- optional -->
+
+### "dictionary.json": Save some JSON to GCS
+
+Imagine a file named something like `dictionary_<experiment_slug>.json` containing objects that look like:
+
+```json
+{
+    "active_hours": {
+        "friendly_name": "Active hours",
+        "description": "Measures the amount of time (in 5-second increments) during which Firefox received user input from a keyboard or mouse. The Firefox window does not need to be focused.",
+        "bigger_is_better": true
+    },
+    "uri_count": {
+        "friendly_name": "URIs visited",
+        "description": "Counts the total number of URIs visited. Includes within-page navigation events (e.g. to anchors).",
```

Comment:
I'm trying to understand the difference between what we would have here vs. what's in `nimbus-shared` e.g.: https://github.com/mozilla/nimbus-shared/blob/main/data/features/picture_in_picture.toml.

If I understand correctly, `nimbus-shared` is no longer the way to add a probeset but rather directly in experimenter (cc @jaredlockhart for this).

In which case, I'm wondering if this could also be included in that part of experimenter when you create the probeset.

I may be misunderstanding and/or missing some information. Perhaps we can clear it up in our Cirrus meeting today. I think this may be related to the question Jared added to our meeting notes.

Reply:
The key difference is that this documents metrics. Experimenter does not know which metrics Jetstream will produce for a probeset and so Experimenter can't document them.

Consumers presently have no way to connect a probeset to the generated metrics -- today, the probeset slug shows up in the prefix of the generated metric name, but I think it's probably healthier to treat slugs as opaque instead of trying to parse them.

Reply:
Ok cool, given our chat in the Cirrus meeting today, it sounds like adding some extra metadata here about which probeset a metric maps to would be helpful as well as whether this metric is "default" for the probeset.

Reply:
Updated with a proposal for this.